src/view/surface.cpp

```diff
@@ -35,23 +35,36 @@ void wf::surface_interface_t::add_subsurface(
     this->emit_signal("subsurface-added", &ev);
 }
 
-void wf::surface_interface_t::remove_subsurface(
+std::unique_ptr<wf::surface_interface_t> wf::surface_interface_t::remove_subsurface(
     nonstd::observer_ptr<surface_interface_t> subsurface)
 {
     auto remove_from = [=] (auto& container)
     {
-        auto it = std::remove_if(container.begin(), container.end(),
+        auto it = std::find_if(container.begin(), container.end(),
             [=] (const auto& ptr) { return ptr.get() == subsurface.get(); });
+
+        std::unique_ptr<surface_interface_t> ret = nullptr;
+        if (it != container.end())
+        {
+            ret = std::move(*it);
+        }
+
         container.erase(it, container.end());
```

Comment:
This would delete all subsurfaces until the end of the vector, right? We definitely do not want this.

Reply:
woops sorry for that. Fixed now.